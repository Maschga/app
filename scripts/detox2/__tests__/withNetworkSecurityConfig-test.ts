import {
  getTemplateFile,
  withNetworkSecurityConfigManifest,
} from "../withNetworkSecurityConfig";

describe(getTemplateFile, () => {
  it(`returns only base-config when subdomains is '*'`, () => {
    const content = getTemplateFile("*");
    expect(content).toMatchSnapshot();
    expect(content).toMatch(/base-config/);
    expect(content).not.toMatch(/domain-config/);
  });
  it(`returns domain-config when subdomains is an array`, () => {
    const content = getTemplateFile(["localhost"]);
    expect(content).toMatchSnapshot();
    expect(content).toMatch(/domain-config/);
    expect(content).toMatch(/localhost/);
  });
});

const exp = { name: "foo", slug: "bar" };

describe(withNetworkSecurityConfigManifest, () => {
  it(`doesn't modify config if subdomains is an empty array`, () => {
    const applied = withNetworkSecurityConfigManifest(exp, {
      subdomains: [],
    });
    expect(applied).toMatchSnapshot();
    expect(applied).toStrictEqual(exp);
  });
  it(`modifies config if subdomains is not empty`, () => {
    const applied = withNetworkSecurityConfigManifest(exp, {
      subdomains: ["10.0.2.2"],
    });
    expect(applied).toMatchSnapshot();
    expect(applied).toHaveProperty("mods.android");
  });
});
