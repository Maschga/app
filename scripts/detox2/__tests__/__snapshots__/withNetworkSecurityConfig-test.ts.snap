// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTemplateFile returns domain-config when subdomains is an array 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<network-security-config>

    <domain-config cleartextTrafficPermitted="true">
      <domain includeSubdomains="true">localhost</domain>
    </domain-config>
  
</network-security-config>"
`;

exports[`getTemplateFile returns only base-config when subdomains is '*' 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
<base-config cleartextTrafficPermitted="true" />
</network-security-config>"
`;

exports[`withNetworkSecurityConfigManifest doesn't modify config if subdomains is an empty array 1`] = `
{
  "name": "foo",
  "slug": "bar",
}
`;

exports[`withNetworkSecurityConfigManifest modifies config if subdomains is not empty 1`] = `
{
  "mods": {
    "android": {
      "dangerous": [Function],
      "manifest": [Function],
    },
  },
  "name": "foo",
  "slug": "bar",
}
`;
